### ISSUE TITLE
Increase text size or change colour

### ISSUE THREAD

**Issue by UKPhil** [ubports/ubuntu-touch#320]
I have ported a OnePlus One over to UBPorts and on the whole it looks and feels great. However, I am finding that the text on the screen is either slightly too small or the colour does not make the wording stand out enough. On pages like Settings, I am finding myself having to squint a bit to see things. I

If you look at the other phones (Android & IOS) their text is generally slightly darker (even black sometimes) which make it easier to read.

I know some will say that it is OK as it is and that I may need glasses, but for me if I look at the Settings page the text size and colour do not make it easy to see. With the World Health Organisation estimating that 253 million people live with vision impairment and 217 million have moderate to severe vision impairment. This is a sector of the market that cannot be ignored, by just saying it is ok for the majority.

**Comment by NeoTheThird** [comment]
You might have a point there. For me personally, i think the text size is ok, but it's definitely not too big. Especially the smaller text on some subpages is ridiculously small. I would personally prefer to make a change of the default text size global though, so we don't sacrifice consistency.

**Comment by ChloeWolfieGirl** [comment]
Personally I like the text size, infact Ive said the opposite in a bug report about indicator text size.

Im someone who likes a lot of data on my screen, and comparing how much I get out of my screen space on my opo compaired to my op3, is no competition, the text used to be so much smaller a year ago, and you had to get close to the screen to seeanything.

After a little while we adjusted the gu sizes, and this is where we've ended up.

If we scale up the gu then a lot of things are to big, the indicators take up a lot more space, many apps dont look as good etc.

With that being said, I can definitely see people struggling with text size, so it needs to be tackled, personally I think we really need to introduce an accessibility page, and in there have a slider for text size.

**Comment by UKPhil** [comment]
Even if we do not increase the text size a module in the settings allowing users to increase if they wish is a good idea. The last thing we want to do is ha e an os that discriminates certain users.

**Comment by ChloeWolfieGirl** [comment]
Completely agree, I also made a bug recently because I have a some friends with dyslexia who find it hard to read black on white text, but iphone has a feature to add colour overlays.

We definitly need this kind of stuff, but implementing it and doing it right is something we need to do.

We shouldn't have the screen automatically red with big text, and we should make sure options are there to have text and colours how ever you need or want them, expecially if you have complications like bad eye site or dyslexia.

In tweak tool theres an option to make icons bigger or smaller in the launcher, we should merge some of those features into the system settings me thinks.

**Comment by ChloeWolfieGirl** [comment]
So I was looking into text size on my android phone, and theres a scaleable option in display, along with text size, perhaps we should have both.

It defaults at medium, giving options to make the text and display really small or really big.

![file1509300850097](https://user-images.githubusercontent.com/8162805/32146873-c4a6a0a0-bcd5-11e7-8dc9-8aba4432010f.png)
![file1509300835254](https://user-images.githubusercontent.com/8162805/32146843-81025b50-bcd5-11e7-936f-b721e46e24db.png)
![file1509300841850](https://user-images.githubusercontent.com/8162805/32146846-87ea56de-bcd5-11e7-97bc-d7e89599c55f.png)
![file1509300831880](https://user-images.githubusercontent.com/8162805/32146851-90874f22-bcd5-11e7-8b92-aa9745914a5e.png)
![file1509300839123](https://user-images.githubusercontent.com/8162805/32146854-94481876-bcd5-11e7-9a64-4fea8b824f5e.png)

**Comment by UKPhil** [comment]
@ChloeWolfieGirl I totally agree and think that we as an os should have both options. The last thing we want to do is force people away who just cannot see the text on our os. I will have a look at tweak tools and their options. However, we should get something  in the settings screen so users do not have to rely on downloading other tools. I agree with what you say

**Comment by ChloeWolfieGirl** [comment]
Personally I don't have an issue with either of those things.

I think the only thing I have issue with related to these are, buttons seem to big on the dialer (The text/numbersare fine but to spread out), and the message for the mouse and touch pad, should be a bit closer to the title.

The message for the mouse and touchpad is the same font size and colour as other parts of the system settings, though perhaps it should be changed to be more similar to the sim menu when you don't have a sim installed?

**Comment by UKPhil** [comment]
The main point of this post it to say that as an os we do not just need to be caring about what the able bodied think, we also need to consider users who may have sight issues. We need to not discriminate with our os and who can use it.

**Comment by ChloeWolfieGirl** [comment]
That's why I suggested we copy the display size and text size options from android.

**Comment by sverzegnassi** [comment]
![screenshot_20171102-145447](https://user-images.githubusercontent.com/1218443/32329695-0b60b930-bfde-11e7-97fd-be3b51eea2a9.png)
![screenshot_20171102-145500](https://user-images.githubusercontent.com/1218443/32329696-0b94e138-bfde-11e7-9f14-e0e2f5efc108.png)

There are two different settings on Android, one for appearances, one for accessibility. Of course, we might prefer to keep just one (the former, already referenced in previous comments), and adopt it also for the latter use case.

Still to be evaluated, how to ensure all the applications should follow user-defined text size. It is not a problem for Ubuntu UI Toolkit apps, as they consistently rely on "abstract" sizes[1] (this is true for core apps - some third party app specify text size in .gu or even .dp).

However I suspect that getting e.g. Libertine apps, or even snap apps, working with such setting might reveal to be trickier, and requires further investigation.

[1] https://github.com/ubports/ubuntu-ui-toolkit/blob/master/src/UbuntuToolkit/ucfontutils_p.h

**Comment by sverzegnassi** [comment]
I believe that UBports wants to design an operative system for everyone, including people with specific conditions or deseases. In these terms, providing accessibility settings should be a priority ;)

@NeoTheThird , what do you think? Could we promote this report from the 'opinion' status?

**Comment by Flohack74** [comment]
Definitely, yes. But we need a good proposal, how this can be inserted at the right place, with dynamic capabilities so that you can see immediately the result. Could be tricky...

**Comment by ChloeWolfieGirl** [comment]
Can't we copy how it works in Gnome, or implement something similar?

Ubuntu looks at the screen, creates default settings, etc.

Then if you install Tweak tools you can change text size for Documents, interface, window title, and monospace, this even lets you change font, and scale the fonts.

This probably wouldn't apply to most apps, even large text doesn't affect all apps.
BUT, we could have it so that apps can look at the setting and if the settings on, the app could up its text size, but then apps would have to consider that setting all the time.


The display size option seems like it'd be more difficult to implement, it wouldn't be in accessability either, it'd be in displays, but it would cover all applications without app devs having to do anything or consider peoples size preferences.

This could be scaled up or down, and even though it wouldn't specifically state its for accessability it would help those who cant read the options.

The negative side I could see from it is, if you make stuff scale to big, some things might not be very receptive to it, if you look at one of the smaller phones such as the N4, if that display is automatically at midway, and you scaled the size up, the indicators would go off screen, meaning you couldn't see or access some of the scopes, I dont think this is much of an issue for bigger display's, but we have to be aware of some issues that may come up.

Of cause none of this answers @Flohack74 's how could we implement it so you can see results immediately, I hope it can help those who can propose a good result, better idea of what things might change, the benifits and negatives of both, etc.

**Comment by UKPhil** [comment]
Love the thoughts @ChloeWolfieGirl

**Comment by UKPhil** [comment]
I have recently downloaded a new file manager, as there are only two available and the one that I have was not working correct. Anyway, in this app there is a great settings section, which among other things give you the option to either have your font size (S, M, L or XL). This works great and when I originally thought of when raising this issue/feature request. Something like this in the settings section for UBPorts would be great.

**Comment by NeoTheThird** [comment]
Which filemanager app are you talking about? Can you provide a link to open.uappexplorer.com and maybe a screenshot?

**Comment by NeoTheThird** [comment]
I don't know if this is technically possible, but couldn't you apply the scale factor to the fontsize as well? That way, if an app has set some text to be 14 px, and a scale factor of 1.2 is used, it shows up as 17 px? Please excuse my noob question.

> My best guess is that we'd do better if we wait for 'xenial', before implementing any disruptive change.

Definitely. This is not that urgent and should be planned well.

**Comment by sverzegnassi** [comment]
@NeoTheThird Whether we set `units.dp(11)` or `units.dp(16)` directly, or we apply a multiplier factor, is mostly equivalent. Perhaps, using dp() or pixels might be preferable, since it's straightforward and would represent a number with a clear meaning.

How we expose the related option in system-settings is a minor concern. We might use a slider that offers a discrete preset of values, in terms of multiplier to a base pixel size, and then store the exact pixel size in the settings backend.

On the other hand, I just realized that we can't get this on ubuntu-ui-toolkit.
I checked how the text size settings work on my iPad, and I see that iOS only tweaks the size for regular text, leaving e.g. titles, actions, and buttons text size untouched. This solution actually makes sense, since UI components usually set their height independently from text.

[[Screenshot] Maximum text size in Files app (iPad)](https://user-images.githubusercontent.com/1218443/33887699-ee08268e-df4a-11e7-8471-e3e1275f3ff0.jpg)

This is something we can't do in ubuntu-ui-toolkit, since we have just a single font size which is shared by all the components. Changing that font size would lead to unpredictable issues with the UX - e.g. we might have button text which is bigger than the button itself. Unusable apps.

For ubuntu-ui-toolkit the only way to keep things predictable is to resize everything accordingly; this can be achieved by making the grid unit configurable (see issue #226), which is equivalent to changing the DPI for a screen.

Instead, QtQuick Controls 2 would allow us to use different font settings for different components, giving us all the flexibility we might need for getting this functionality fully working.

**Comment by WLBI** [comment]
Hey wow, see this: **Making Ubuntu Touch readable**
https://andybleadenubuntutouch.blogspot.de/
That's a howto, for making bigger font size in Ubuntu Touch.
This compared with something like this: https://user-images.githubusercontent.com/1218443/32329695-0b60b930-bfde-11e7-97fd-be3b51eea2a9.png
would be perfect!

**Comment by UKPhil** [comment]
@WLBI thanks for your reply. I have since followed the method in the user manual and have been able to increase my text and fonts. I also can now read what my phone is telling me. The only issue with doing it this way is that the notifications are not increased in size. A small price to pay. 

A system setting of this would be fantastic.

**Comment by UKPhil** [comment]
@WLBI thanks for your reply. I have since followed the method in the user manual and have been able to increase my text and fonts. I also can now read what my phone is telling me. The only issue with doing it this way is that the notifications are not increased in size, but all the other items are increased in the panel. A small price to pay.

A system setting of this would be fantastic.

**Comment by UKPhil** [comment]
Last night I increased the size of my text on my M10 using the method above. So now I have both my tablet and OnePlus One with increased text size. All i can say is wow what a difference. Now I can see the text so much clearer and it is a pleasure to use the tablet. I showed my wife and she immediately asked me to do the same to her M10 running UBPorts. Guys we really need to get this in a release soon.

**Comment by WLBI** [comment]
@UKPhil 
That's what I already like to have years back, when canonical bring the BQ M4.5 as the first Ubuntu Touch device. I asked for that feature many times, but nobody was interested in that, because they all have eagle eyes ;-)

**Comment by UKPhil** [comment]
@WLBI lets hope that the guys at UBPorts do hear us and that this is finally actioned. From what I have seen so far they do seem to be listening to us, which is good.

**Comment by UKPhil** [comment]
@WLBI I have been using the method, as advised in your post above now for the last few weeks. All is good at the start and it is nice to see the font nice and clear for a change. Deferentially something that UT would benefit from. I have noticed one small issue though. The browser does not seem to be able to operate with the size settings. The screen is displayed correctly and nice and big, but when you click on something it is either not working correct or taking you to another page. The screen is larger, but it still seems to be operating on the older smaller size setting.

**Comment by abkro** [comment]
@UKPhil @WLBI and the rest who have been keeping this discussion alive: I have +1.5 vision on both eyes, which is common in the general population, and I can't read SMS messages on my devices (Nexus 5 and bq E4.5) without glasses. I can barely make out the names in the Messaging app main view. I sorely need this to be fixed, but with the current state of 16.04, I can see there are more important issues just now, Maybe it can be pushed up the list as soon as 16.04 is released, and then include it in an update?

**Comment by UKPhil** [comment]
@abkro thank you for your comments and helping to keep this topic alive. I totally agree with version 16.04 taking the priority. However, for now I think that changing the colour of the text  may be a bit of an interim fix until the text size can be addressed. This would help some users.

**Comment by dr47** [comment]
Thank you for instructions to the font size guide (https://andybleadenubuntutouch.blogspot.com/2018/01/sorting-font-size-out.html).
This makes Ubuntu Touch usable for me, WOW!

I suggest that this important setting (for me and people similar to me) be made more visible:

- Write these instructions in a more visible place (release notes, FAQ?)
- Make GUI workaround in System setting => to create .config/upstart/gu.conf" easier.

This would save time (and eyes!) for people who need a more accessible interface.
A dark theme is another important thing for a comfortable use.

Thank developers for great work (Thanks also through PAYPAL :-) and sorry for my english.

**Comment by UKPhil** [comment]
I totally agree.

**Comment by doniks** [comment]
docs.ubports.com welcomes contributions

**Comment by Eupalino** [comment]
Bump.

I couldn't agree more. Ubports should have S M L text size choice like Android has.

**Comment by UKPhil** [comment]
Let's only hope that UBports take notice of what their user Base are saying and implement this. As time goes by I feel the need for more and more bigger text. It is OK for younger people to say it is OK but their views will change when they get older and sight goes slightly. Also there is a vast about of younger people with sight issues. By not implementing this we are excluding them from our user group.

Please please let's get some movement on this soon.

**Comment by dobey** [comment]
I don't think S/M/L options will be doable really, but whole UI scale should be doable (though apps which specify font size using hard-coded pixel or point values may still be problematic).

And please don't use such threatening/condescending tones when discussing the bugs. It doesn't help, and making yourself out as some sort of victim by suggesting developers aren't listening to users, isn't going to get issues resolved any faster either.

**Comment by mateosalta** [comment]
I like the smaller element and font size, but I have to agree that in user testing with my parents they often comment on the small size. being able to change this quickly to suit would be a great option.

**Comment by Vistaus** [comment]
I support this. I like the smaller element size, but the font size is just too small for my eyes. So an option to set the font size would be great (UT Tweak Tool allows to increase it a bit, but it also increases element size, which I don't like).
